src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java

```diff
@@ -69,6 +70,10 @@ public KNN80DocValuesProducer(DocValuesProducer delegate, SegmentReadState state
             if (!field.attributes().containsKey(KNN_FIELD)) {
                 continue;
             }
+            // Only segments that contains BinaryDocValues and doesn't have vector values
+            if( field.getDocValuesType() != DocValuesType.BINARY || field.hasVectorValues()){
```

Comment:
why would this happen both times?

Reply:
@jmazanec15 I updated comment. Let me know if that is not clear or missing some more context

Reply:
I guess I was thinking its redundant if we already check the KNN_FIELD attribute